modules/mobile/app/models/mobile/v0/adapters/service.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'common/models/base'
+require 'date'
+
+module Mobile
+  module V0
+    module Adapters
+      class Service
+        # 411 = Podiatry
+        SERVICE_TYPE_IDS = %w[amputation audiology covid optometry outpatientMentalHealth moveProgram foodAndNutrition
+                              clinicalPharmacyPrimaryCare 411 primaryCare homeSleepTesting socialWork].freeze
+
+        def parse(service_eligibilities)
+          SERVICE_TYPE_IDS.collect do |service|
+            request_facilities = []
+            direct_facilities = []
+
+            service_eligibilities.each do |facility|
+              facility_id = facility.facility_id
+              facility_service = facility.services.select { |h| h[:id] == service }.first
+              next if facility_service.nil?
+
+              request_facilities << facility_id if facility_service.dig(:request, :enabled)
+              direct_facilities << facility_id if facility_service.dig(:direct, :enabled)
+            end
+
+            request_facilities.compact!
```

Comment:
Why do we have to compact? Is it possible to have nil facility ids?

Reply:
nope. Left over from the previous way I wrote this.